h2o-algos/src/main/java/hex/kmeans/KMeansModel.java

```diff
@@ -70,7 +68,17 @@
     public KMeansOutput( KMeans b ) { super(b); }
   }
 
-  public KMeansModel(Key selfKey, KMeansParameters parms, KMeansOutput output) { super(selfKey,parms,output); }
+  public KMeansModel(Key selfKey, KMeansParameters parms, KMeansOutput output) { 
+    super(selfKey,parms,output);
+  }
+
+  @Override
+  public void computeEffectiveParameters() {
+    _effective_parms._distribution = null;
```

Comment:
super?

Reply:
similarly, super would set distribution, stopping_metric, fold_assignment and categorical encoding to null, and then the rest after super would set fold_assignment and categorical encoding to effective value. distribution and stopping_metric are not used in KMeans, they are not returned on clients as parametes, but in this stage we have them present. hence they are set to null. Otherwise, they would be = AUTO and wouldn't pass through checkEffectiveParmsDoesNotContainAuto()